presto-main/src/main/java/com/facebook/presto/operator/LookupSourceFactory.java

```diff
@@ -28,6 +28,11 @@
 
     ListenableFuture<LookupSource> createLookupSource();
 
+    /**
+     * Can be called only after {@link #createLookupSource()} is done and all users of {@link LookupSource}-s finished.
```

Comment:
Done. For non-partitioned LS it was already done in `com.facebook.presto.operator.OuterLookupSource.OuterPositionTracker#finished`. For partitioned i added check in a fixup.